engines/engine.h

```diff
@@ -96,31 +134,58 @@ class PauseToken {
 	PauseToken(Engine *);
 
 	Engine *_engine;
-
+	/**
+	 * @see Engine
+	 */
 	friend class Engine;
 };
 
 class Engine {
 public:
+	/**
+	 * The OSystem instance used by the engine.
+	 */
 	OSystem *_system;
+	/**
+	 * The Mixer instance used by the engine.
+	 */
 	Audio::Mixer *_mixer;
 
 protected:
+	/**
+	 * The TimerManager instance used by the engine.
+	 */
 	Common::TimerManager *_timer;
+	/**
+	 * The EventManager instance used by the engine.
+	 */
 	Common::EventManager *_eventMan;
+	/**
+	 * The SaveFileMAnager used by the engine.
+	 */
 	Common::SaveFileManager *_saveFileMan;
 
+	/**
+	 * The Dialog instance used by the engine.
+	 */
 	GUI::Dialog *_mainMenuDialog;
+	/**
+	 * Run the main menu dialog.
```

Comment:
Run a GUI dialog. This can be the main menu dialog, but this can also be a different dialog.
To show the main menu dialog, you would not use this function though. You would call `openMainMenuDialog()` instead.